bliss.shy.js

```diff
@@ -514,11 +512,16 @@ $.Element.prototype = {
 					transitionProperty: Object.keys(props).join(", ")
 				});
 
-				$.once(this, "transitionend", function() {
-					clearTimeout(i);
-					$.style(this, previous);
+				if (duration === 0) {
 					resolve(this);
-				});
+				}
+				else {
+					$.once(this, "transitionend", function() {
+						clearTimeout(i);
+						$.style(this, previous);
+						resolve(this);
+					});
+				}
 
 				// Failsafe, in case transitionend doesn’t fire
 				var i = setTimeout(resolve, duration+50, this);
```

Comment:
The failsafe is not needed if duration is 0, it should go inside the else. Or just `return resolve(this)` above so that you don't need the else anyway.